### ISSUE TITLE
Animations - Explore Rive animations

### ISSUE THREAD

**Issue by J-Son89** [status-im/status-react#18193]
The wallet send screens ideal state are planning to include animations with multiple outcomes, for animations like this it can work well to use Rive. 

Do a spike on whether it this package will work for us or not.
-> i.e add this package to our codebase,
->  have a test animation that has two end states triggered by some event 
-> ensure working on ios and android 
-> okay if it only works on more powerful phones as we can whitelist these.

Document your findings in Notion or a GH issue/discussion etc.

Links
https://github.com/rive-app/rive-react-native

see: https://help.rive.app/runtimes/overview/react-native

**Comment by shivekkhurana** [comment]
I've started with ios as that's the device I have rn. Will look at simulators for Android. 
Today's task is to learn Rive.

**Comment by shivekkhurana** [comment]
While setting things up, I realised that Rive's [minimum deployment target is ios 14](https://help.rive.app/runtimes/overview/react-native#overview). We are supporting as low as 12.4.

I'm updating the base to 14 to see if anything major breaks down, and if I can get Rive to work.

**Comment by J-Son89** [comment]
Forget to mention, we discussed this and we are okay if we need to make that change 👍

**Comment by shivekkhurana** [comment]
After fighting with updating deployment base to 15, I was greeted with this wild error. 
Have been stuck on it for a while now. Taking a break for lunch.

<details><summary>📷 Error Screenshot</summary>
<p>

![Simulator Screenshot - iPhone 13 - 2023-12-18 at 13 25 05](https://github.com/status-im/status-mobile/assets/1925158/962503f7-974f-459a-948c-5ac3ab4db07b)

</p>
</details>

**Comment by J-Son89** [comment]
I had this commit, perhaps worth checking 
https://github.com/status-im/status-mobile/compare/develop...jc/wallet-animation

**Comment by shivekkhurana** [comment]
Got it up. Didn't need to mess with XCode deployment target, not Pods deployment target. Only the `min_ios_deployment` version as suggested by JC. 

<details><summary>📼 Here's what it looks like</summary>
<p>


https://github.com/status-im/status-mobile/assets/1925158/d3639c19-5be4-4a0d-9f8a-df9fd2d5ca8c



</p>
</details>

**Comment by J-Son89** [comment]
@shivekkhurana - shall we close this issue? Perhaps we record the findings in Notion? 🤔

**Comment by shivekkhurana** [comment]
I'm fine with Notion or Github. 

Preliminary Report

Objective: To score performance of Rive motion fx agains mp4 videos.

First I opened the enable biometrics screen and recorded rendering time using flame graphs (with mp4 video).
- The video took 10.7 ms to render.
- There were multiple steps in rendering due to the parallax effect and muliple videos being layered. Graph attached.
- Total size of mp4 videos is approx. 9mb
- In total, the component took 21.9 ms to render. 
Here's the flame graph 
<img width="1166" alt="Screenshot 2024-02-21 at 2 37 58 PM" src="https://github.com/status-im/status-mobile/assets/1925158/d56f64ca-a99c-4cff-9465-bf6b562a09d0">


Then I replaced the video with the Rive animation, and repeated the same test.

- The Rive animation appeared to be visually more pleasant (could be placebo)
- The component took 3.1 ms to render 💨
- Overall render time was 34 ms because the buttons took more time to render
- Total size of the rive file is 13.9mb (with SVGs, so no scope for improvement here)
- I haven't connected parallax yet, working on it.
<img width="1517" alt="Screenshot 2024-02-21 at 2 39 24 PM" src="https://github.com/status-im/status-mobile/assets/1925158/ef5946d0-be51-421f-b950-9b13fc91a93c">

TL;DR;  Rive is much better in terms of perf and render time. But costs about 50% more in bundle size.

Here are videos of both, I can't tell any difference in fluidity. The only difference is the difference in animations:

With mp4

https://github.com/status-im/status-mobile/assets/1925158/f8f3bb67-e8ee-40fc-bb52-e9316d3c3a18 

With Rive

https://github.com/status-im/status-mobile/assets/1925158/fe0fd871-1be1-4068-a8e5-21b6baf72f6f

**Comment by shivekkhurana** [comment]
Here's a version with the assets sizes reduced. On video (and visually), I can see no difference.

https://github.com/status-im/status-mobile/assets/1925158/7d90f94c-abc4-41a3-8bc7-51c39d3a32f1

**Comment by shivekkhurana** [comment]
We need to restrict the parallax. And not use pointer events for parallax, instead accept the state of the phone's gyro as input:

https://github.com/status-im/status-mobile/assets/1925158/5f521904-05b7-44fb-aa78-37625301228c

**Comment by J-Son89** [comment]
We can close this issue for now, as Rive explorations have stopped for the foreseeable future.
